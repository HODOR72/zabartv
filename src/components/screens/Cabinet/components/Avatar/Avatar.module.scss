.avatar {
	display: flex;
	align-items: center;
	justify-content: center;
	font-weight: 700;
	font-size: 64px;
	line-height: 80px;
	width: 162px;
	height: 162px;
	flex-shrink: 0;
	border-radius: 100%;
	background-color: #13213c;
	color: #fff;

	@media (max-width: 1200px) {
		width: 140px;
		height: 140px;
		font-size: 50px;
		line-height: 65px;
	}

	@media (max-width: 768px) {
		width: 130px;
		height: 130px;
	}

	@media (max-width: 576px) {
		width: 117px;
		height: 117px;
		font-size: 34px;
		line-height: 43px;
	}
}
