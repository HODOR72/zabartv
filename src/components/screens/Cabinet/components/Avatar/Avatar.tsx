import { FC } from 'react';
import styles from './Avatar.module.scss';

interface AvatarProps {
	name: string | undefined;
}

export const Avatar: FC<AvatarProps> = ({ name }) => {
	/* Временно */
	const letter = name ? name[0] : null;

	return (
		<div data-testid="avatar" className={styles.avatar}>
			{letter}
		</div>
	);
};
