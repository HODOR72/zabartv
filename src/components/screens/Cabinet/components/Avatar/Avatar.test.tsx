import { render, screen, fireEvent } from '@testing-library/react';
import { Avatar } from './Avatar';

describe('Avatar.test.tsx', () => {
	test('tests', () => {
		render(<Avatar name="First Last" />);
		const element = screen.getByTestId('avatar');

		expect(element).toHaveTextContent('F');

		expect(element).toMatchSnapshot();
	});
});
