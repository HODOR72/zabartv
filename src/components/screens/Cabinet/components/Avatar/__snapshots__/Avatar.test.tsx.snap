// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar.test.tsx tests 1`] = `
<div
  class="avatar"
  data-testid="avatar"
>
  F
</div>
`;
